gst-libs/mfx/gstmfxallocator_d3d11.c

```diff
@@ -49,7 +49,7 @@ gst_mfx_task_frame_alloc(mfxHDL pthis, mfxFrameAllocRequest * request,
   if (task->task_type & (GST_MFX_TASK_VPP_OUT | GST_MFX_TASK_DECODER))
   request->Type |= 0x1000;*/
 
-  if (priv->saved_responses && priv->task_type & GST_MFX_TASK_DECODER) {
+  if (priv->saved_responses && (priv->task_type & GST_MFX_TASK_DECODER) == GST_MFX_TASK_DECODER) {
```

Comment:
I'm OK with the change for clarity. Also mfxvpp not rendering to video memory seems to not be the root cause, but the decoder has GstMfxFilter to do color space conversion to BGRA, and it works. What happened?

Yes, the decoder invokes this allocate method implicitly when it is initialized (MFXVideoDECODE_Init()) so when the filter is being shared with decoder and the shared surfaces have already been allocated, this block is to ensure that the already successful allocation response is indicated to the decoder, so that there is no need to allocate the surfaces again.